pandas/tests/indexes/conftest.py

```diff
@@ -28,6 +29,31 @@ def indices(request):
     return request.param
 
 
+def _make_repeating_index(x=10):
+    # x should be > 1
+    return Index(sorted([i for i in range(x // 2 + 1)] * 2)[:x])
+
+
+@pytest.fixture(params=[tm.makeUnicodeIndex,
```

Comment:
doesn't the indices fixture work?

Reply:
I can't remember the exact reasoning at this point, I have to go back and check. But I think it mostly had to do with the fact that since the Indices are pre-instantiated, you would get pairs like this:

    >>> idx1 = tm.makePeriodIndex()
    >>> idx2 = tm.makeDateIndex()
    >>> idx1.equals(idx2)
   True

And when the indexes are equal a different logic path is taken. However, now that I'm testing it out, I think that some things have changed (like doing the incompatibility check before checking if the indexes are equal) that might make this not an issue anymore.

I remember there being more examples of this (**these are probably bugs that I should have reported, will see if I can find more examples**). And this was causing the tests to not be as thorough as I wanted, so being able to instantiate them with different sizes ensured that this would never happen.

In any case I will open an issue for the equals bug and try out the original fixture again.